### PR TITLE
enable pairing with branches from ksctl repository

### PR DESCRIPTION
**Pull request by MatousJobanek**
Since the e2e test setup is increasingly dependent on ksctl commands, I introduced a possibility of pairing between toolchain-e2e and ksctl.
* It's only one-directional  - the e2e tests are not executed in ksctl repo for now. It can be changed later.
* It's not included into the limit that we can pair only one PR (host/member/reg-service) with only one PR from toolchain-e2e repo. In other words, it can be combined with it.
* it's not included in the migration test suite for now
* local execution uses ksctl version from master if not specified otherwise via env variables `USE_INSTALLED_KSCTL` or `KSCTL_REPO_PATH`. The UX can be changed later.

paired PR https://github.com/kubesaw/ksctl/pull/58 (expecting to fail when generating tier manifests)

### CODE REVIEW

File: make/ksctl.mk [hunk 1]
```diff
@@ -1,7 +1,62 @@
...
+			git config --global user.email "devtools@redhat.com"
+			git config --global user.name "Devtools"
+			# clone
+			git clone --depth=1 https://github.com/kubesaw/ksctl.git ${KSCTL_REPO_PATH}
+			# add the user's fork as remote repo
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} remote add external ${AUTHOR_LINK}/ksctl.git
+			# fetch the branch
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} fetch external ${REMOTE_KSCTL_BRANCH}
+			# merge the branch with master
+			git --git-dir=${KSCTL_REPO_PATH}/.git --work-tree=${KSCTL_REPO_PATH} merge --allow-unrelated-histories --no-commit FETCH_HEAD
```
Comment (mfrancisc) [inline make/ksctl.mk:49]: just curious - why do we need to clone both the upstream and the fork and merge them ? was there any problem with cloning what was specified in the `KSCTL_GH_OWNER` variable and `KSCTL_GH_BRANCH` and building and installing that ?

Comment (mfrancisc) [reply make/ksctl.mk:49]: also could there be merge conflicts when doing this ?

Comment (mfrancisc) [reply make/ksctl.mk:49]: Ok now I see , I think we want it to work in the same way as the pairing for the other repos, so if there is a conflict the author must merge/fix and then rerun the tests. 

Not sure why - but I was thinking that initially we would have a simpler mechanism that just clones a given ksctl repo , builds and installs it.

**Review by alexeykazakov — APPROVED** [review]
:+1:

File: make/ksctl.mk (hunk 1 above)
Comment (MatousJobanek) [reply make/ksctl.mk:49]: we want to test the final state - when the PR is merged to master. We don't want test only the changes in the PR (without combining them with master) as the PR may be outdated and it doesn't have to contain the latest changes important for the functions of the commands.

**Comment by MatousJobanek** [comment]
/retest
the paired PR was closed and the branch was deleted

**Review by mfrancisc — APPROVED** [review]
Looks great 👍  

Thanks a lot for creating this and for answering my comments! 

minor note -  maybe we could document somewhere these variables:
`USE_INSTALLED_KSCTL=true` would use the installed ksctl version
`KSCTL_REPO_PATH=/my/local/path/to/ksctl/` would build your local version of ksctl 


WDYT?

**Pull request merged**